#163114860 Create Report Model

Easybuoy opened the pull request:
#### What does this PR do?
Create a Report Model and Migration

#### Description of Task to be completed?
- Create `report` migration flie in `migration` directory and add the migrations.
- Create `report` model file in `model` directory and add the model.

#### How should this be manually tested?
- run `npm run db:migrate`
- check `database` to see `report` table amongst table created.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#163114860

#### Screenshots (if appropriate)

#### Questions:

Easybuoy committed:
feat(migration)
- created migration file and added migrations for report table
- created report model
[Delivers #163114860]

oreoluwade commented on models/report.js:
```diff
@@ -0,0 +1,24 @@
+
+    },
+  });
+    // Report.associate = models => Report.hasOne(models.Profile, {
```
Why are these parts commented?

oreoluwade replied:
I imagine there's an association between The `user and the report` as well as between the `article and the report`, at least based on the `userId` and `articleId` fields on the report model. Those relationships are not reflected here. Is there a reason why?

Easybuoy replied:
> Why are these parts commented?

The comment was a mistake, I would remove them.

Easybuoy replied:
> I imagine there's an association between The `user and the report` as well as between the `article and the report`, at least based on the `userId` and `articleId` fields on the report model. Those relationships are not reflected here. Is there a reason why?

The relationships are:
`USER -> HAS MANY -> REPORTS`
`ARTICLE -> HAS MANY -> REPORTS`
So i thought they would have to be defined in the `user` and `article` models and not in the `report` model. That was why I did not define the associations in the `report` model.

oreoluwade replied:
Okay. You can add those relationships in the already existing related files then. 

Also, I'd like you to help change the structure of our migration files naming.

replace the numerical parts with `001`, `002` and so on till you get to the report migration file which will be `006`

Easybuoy replied:
I've made requested changes and added the relationships.

oreoluwade replied:
Okay.

The commented code is still in the report model however.

Easybuoy replied:
Forgot again, removed and pushed.

Easybuoy replied:
Travis failed, I am trying to find out why.

oreoluwade replied:
It seems the `Reports` table is not being created when migrations are done.

oreoluwade replied:
> It seems the `Reports` table is not being created when migrations are done.

This was a test-related thing. Disregard.

Easybuoy replied:
The `Reports` table is created when migrations are done.
I traced the travis error to the relationship i defined in the user model.
`User.hasMany(models.Report, {`
      `foreignKey: 'userId',`
      `onDelete: 'CASCADE'`
    });`

Easybuoy committed:
feat(migration)
- renamed all migrations to use 3 digit numbers
added associations in user and article model
[Delivers #163114860]

Easybuoy committed:
feat(migration)
- removed comment in report model
[Delivers #163114860]

oreoluwade reviewed (dismissed):
LGTM

oreoluwade commented:
Meanwhile, please add these two scripts to package.json:
 `"db:migrate:testdb": "NODE_ENV=test sequelize db:migrate",`
 `"undo:migrations:testdb": "NODE_ENV=test sequelize db:migrate:undo:all"`

Easybuoy commented:
> Meanwhile, please add these two scripts to package.json:
> `"db:migrate:testdb": "NODE_ENV=test sequelize db:migrate",`
> `"undo:migrations:testdb": "NODE_ENV=test sequelize db:migrate:undo:all"`

Okay

oreoluwade commented on models/report.js:
```diff
@@ -0,0 +1,19 @@
+      allowNull: false,
+    },
+    userId: {
+      type: DataTypes.STRING,
```
This is where the problem is. The `userId` type should be `INTEGER`, not `STRING`

Easybuoy replied:
wow, thanks.

Easybuoy replied:
I've pushed the correction

oreoluwade commented on models/report.js:
```diff
@@ -0,0 +1,19 @@
+    articleId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+
```
Remove this empty line.

Easybuoy replied:
Done

Easybuoy committed:
feat(migration)
- changed userId dataType to INTEGER in report model
- added db:migrate:testdb and undo:migrations:testdb scripts in package.json
[Delivers #163114860]

Easybuoy committed:
feat(migration)
- removed empty line
[Delivers #163114860]
